camera.js

```diff
@@ -3623,29 +3643,30 @@ s.auth=function(params,cb,res,req){
 }
 //super user authentication handler
 s.superAuth=function(x,callback){
-    req={};
-    req.super=require(location.super);
-    req.super.forEach(function(v,n){
-        if(x.md5===true){
-            x.pass=s.md5(x.pass);
-        }
-        if(x.mail.toLowerCase()===v.mail.toLowerCase()&&x.pass===v.pass){
-            req.found=1;
-            if(x.users===true){
-                s.sqlQuery('SELECT * FROM Users WHERE details NOT LIKE ?',['%"sub"%'],function(err,r) {
-                    callback({$user:v,users:r,config:config,lang:lang})
-                })
-            }else{
-                callback({$user:v,config:config,lang:lang})
+    var superConf = require(location.super);
+    for (var userConf of superConf) {
+        if (x.mail.toLowerCase() === userConf.mail.toLowerCase()) {
+            var valid = false;
+            if (x.directHashComparison && x.pass === userConf.pass) { // TODO this is terrible and needs to be removed
+                valid = true;
+            } else if (bcrypt.compareSync(x.pass, userConf.pass)) { //TODO make async
+                valid = true;
+            }
+            if (valid) {
+                if (x.users === true) {
+                    // Why the hell is a list of users being returned from an auth function???
```

Comment:
LOL because it's the superuser (default :admin@shinobi.video). They are given a list of all the users to review when they log in. I could make it a separate API call but this was easier when i first made it.

This user should never be accessed unless you need to modify configuration for "admin" accounts.

This login can be disabled completely if you delete super.json.

Reply:
please explain?

Reply:
Separation of concerns, an auth function should return true or false, it shouldn't grab data for the admin when it's successful.

Reply:
thats not what i was asking you to explain.. These are :

> You still have to grab extra data to return to the client here anyway,

it is being passed for use in the super user front end? or does your super user panel show you no admin accounts?

> and it doesn't look like it's being used at all here.

because it's not needed there. That is for the websocket authorization.

in the future superuser will have more RESTful API calls and the panel that uses this data will get it that way.

Reply:
My point is that `superAuth` is only being called in 2 places. One of those places, as you note, doesn't need the users list (and now that I'm looking at it again I see it doesn't request it, so it's good that it doesn't do the sql query for nothing).
The other place it's called uses the list of users, but it also does another query to get logs, and reads a config file to return as well. So why not have it there, next to the other query, instead of in the auth function?

Reply:
you're right, when i first made the superuser panel this was how i wrote it. You're looking at some very old code when i had less understanding about how i should write the app i was trying to make.

The logs were added later and added "more properly". I didn't change the user call at the time because it wasn't a pressing issue. It already worked.

but as I already mentioned.

> in the future superuser will have more RESTful API calls and the panel that uses this data will get it that way.